component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -163,9 +173,24 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
         }
         String retryParam = "";
         try {
-            FederatedAuthenticatorUtil.setUsernameFromFirstStep(context);
-            username = String.valueOf(context.getProperty("username"));
-            authenticatedUser = (AuthenticatedUser) context.getProperty("authenticatedUser");
+            if (TOTPUtil.isLocalUser(context)) {
+                FederatedAuthenticatorUtil.setUsernameFromFirstStep(context);
+                username = String.valueOf(context.getProperty("username"));
+                authenticatedUser = (AuthenticatedUser) context.getProperty("authenticatedUser");
+            } else {
+                isFederatedUser = true;
+                Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
+                for (StepConfig stepConfig : stepConfigMap.values()) {
+                    authenticatedUser = stepConfig.getAuthenticatedUser();
+                    if (authenticatedUser != null && isFederatedUser(authenticatedUser) &&
+                            stepConfig.isSubjectAttributeStep()) {
+                        username = authenticatedUser.getUserName();
+                        userId = getFederatedUserId(authenticatedUser, context, stepConfig, tenantDomain);
+                        context.setProperty(TOTPAuthenticatorConstants.FEDERATED_USER_ID, userId);
+                        break;
+                    }
+                }
+            }
```

Comment:
Do a backend validation in case of configured TOTP only. That case **userId** will become null. So, Throw an error to avoid null pointers in authenticator level.

Reply:
Before getting the userId, it will check whether the authenticated user is null or a federated user. Therefore it will not get the user id and throw an error from here[1] since the authenticated user is null if TOTP is only configured.

[1] https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/94/files#diff-8d5c29187e52b2268d09add06e3537a10e090368a0b9e48583079ea098a5eea0L170-L172